kitsune/messages/utils.py

```diff
@@ -65,3 +68,41 @@ def send_message(to, text=None, sender=None):
 def unread_count_for(user):
     """Returns the number of unread messages for the specified user."""
     return InboxMessage.objects.filter(to=user, read=False).count()
+
+
+def create_suggestion(item):
+    """Create a dictionary object for the autocomplete suggestion."""
+    is_user = isinstance(item, User)
+    return {
+        "type": "User" if is_user else "Group",
+        "type_icon": webpack_static(
+            settings.DEFAULT_USER_ICON if is_user else settings.DEFAULT_GROUP_ICON
+        ),
+        "name": item.username if is_user else item.name,
+        "type_and_name": f"User: {item.username}" if is_user else f"Group: {item.name}",
+        "display_name": item.profile.name if is_user else item.name,
+        "avatar": (
+            profile_avatar(item, 24) if is_user else webpack_static(settings.DEFAULT_AVATAR)
+        ),
+    }
+
+
+def find_users_and_groups_by_search(pre, show_groups=False, exact=False):
```

Comment:
Since both current use cases -- the API and the `new_message` view -- need suggestions rather than DB objects, and there's no current use case for DB objects, I would have this function return a list of suggestions by default, and perhaps add a keyword argument that allows the caller to ask for a list of DB objects instead.

Do you think the limit of users or groups -- in this case 10 -- should be configurable via a keyword argument as well?